### PR TITLE
[GraphBolt] Update `CPUCachedFeature` docstring.

### PR DESCRIPTION
**Pull request by mfbalin**
## Description
@Liu-rj FYI. @frozenbugs what do you think about this limitation?

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by Liu-rj** [comment]
How exactly do users know what is the cache size lower bound that they should set? I tried with 1GB CPU cache for products and arxiv but both is still stuck at the first iteration. Is even 1GB not enough?

**Comment by mfbalin** [comment]
> How exactly do users know what is the cache size lower bound that they should set? I tried with 1GB CPU cache for products and arxiv but both is still stuck at the first iteration. Is even 1GB not enough?

How much space do products features take?

**Comment by Liu-rj** [comment]
Products feature takes 935MB. I also tried to set the cache size to be 0.5GB but the process is still stuck. When I turn off `overlap_feature_fetch` option, the program gets segmentation fault (core dumped).

```
python /home/ubuntu/dgl/examples/graphbolt/disk_based_feature/node_classification.py --gpu-cache-size-in-gigabytes=0 --cpu-cache-size-in-gigabytes=0.5 --dataset=ogbn-products --epochs=3

/home/ubuntu/miniconda3/envs/graphbolt/lib/python3.11/site-packages/dgl-2.4-py3.11-linux-x86_64.egg/dgl/graphbolt/__init__.py:17: GBWarning: 
An experimental feature for CUDA allocations is turned on for better allocation
pattern resulting in better memory usage for minibatch GNN training workloads.
See https://pytorch.org/docs/stable/notes/cuda.html#optimizing-memory-usage-with-pytorch-cuda-alloc-conf,
and set the environment variable `PYTORCH_CUDA_ALLOC_CONF=expandable_segments:False`
if you want to disable it.

  gb_warning(CUDA_ALLOCATOR_ENV_WARNING_STR)
Namespace(epochs=3, lr=0.001, num_hidden=256, dropout=0.2, batch_size=1024, num_workers=0, dataset='ogbn-products', root='datasets', fanout='10,10,10', mode='pinned-pinned-cuda', layer_dependency=False, batch_dependency=1, cpu_feature_cache_policy=None, cpu_cache_size_in_gigabytes=0.5, gpu_cache_size_in_gigabytes=0.0, early_stopping_patience=25, sample_mode='sample_neighbor', precision='high', enable_inference=False)
Training in pinned-pinned-cuda mode.
Loading data...
The dataset is already preprocessed.
/home/ubuntu/miniconda3/envs/graphbolt/lib/python3.11/site-packages/dgl-2.4-py3.11-linux-x86_64.egg/dgl/graphbolt/impl/torch_based_feature_store.py:524: GBWarning: `DiskBasedFeature.pin_memory_()` is not supported. Leaving unmodified.
  gb_warning(
Training: 0it [00:00, ?it/s][1]    891598 segmentation fault (core dumped)  python node_classification.py --gpu-cache-size-in-gigabytes=0   --epochs=3
```

**Comment by mfbalin** [comment]
> Products feature takes 935MB. I also tried to set the cache size to be 0.5GB but the process is still stuck. When I turn off `overlap_feature_fetch` option, the program gets segmentation fault (core dumped).
> 
> ```
> python /home/ubuntu/dgl/examples/graphbolt/disk_based_feature/node_classification.py --gpu-cache-size-in-gigabytes=0 --cpu-cache-size-in-gigabytes=0.5 --dataset=ogbn-products --epochs=3
> 
> /home/ubuntu/miniconda3/envs/graphbolt/lib/python3.11/site-packages/dgl-2.4-py3.11-linux-x86_64.egg/dgl/graphbolt/__init__.py:17: GBWarning: 
> An experimental feature for CUDA allocations is turned on for better allocation
> pattern resulting in better memory usage for minibatch GNN training workloads.
> See https://pytorch.org/docs/stable/notes/cuda.html#optimizing-memory-usage-with-pytorch-cuda-alloc-conf,
> and set the environment variable `PYTORCH_CUDA_ALLOC_CONF=expandable_segments:False`
> if you want to disable it.
> 
>   gb_warning(CUDA_ALLOCATOR_ENV_WARNING_STR)
> Namespace(epochs=3, lr=0.001, num_hidden=256, dropout=0.2, batch_size=1024, num_workers=0, dataset='ogbn-products', root='datasets', fanout='10,10,10', mode='pinned-pinned-cuda', layer_dependency=False, batch_dependency=1, cpu_feature_cache_policy=None, cpu_cache_size_in_gigabytes=0.5, gpu_cache_size_in_gigabytes=0.0, early_stopping_patience=25, sample_mode='sample_neighbor', precision='high', enable_inference=False)
> Training in pinned-pinned-cuda mode.
> Loading data...
> The dataset is already preprocessed.
> /home/ubuntu/miniconda3/envs/graphbolt/lib/python3.11/site-packages/dgl-2.4-py3.11-linux-x86_64.egg/dgl/graphbolt/impl/torch_based_feature_store.py:524: GBWarning: `DiskBasedFeature.pin_memory_()` is not supported. Leaving unmodified.
>   gb_warning(
> Training: 0it [00:00, ?it/s][1]    891598 segmentation fault (core dumped)  python node_classification.py --gpu-cache-size-in-gigabytes=0   --epochs=3
> ```

Ok, let me fix the segmentation fault as well.

File: python/dgl/graphbolt/impl/cpu_cached_feature.py [hunk 1]
```diff
@@ -25,7 +25,10 @@ class CPUCachedFeature(Feature):
     fallback_feature : Feature
         The fallback feature.
     max_cache_size_in_bytes : int
-        The capacity of the cache in bytes.
+        The capacity of the cache in bytes. The size should be a few factors
+        larger than the size of each read request. Otherwise, the caching policy
+        will hang due to all cache entries being read and/or write locked,
+        resulting in a deadlock.
```
Comment (frozenbugs) [inline python/dgl/graphbolt/impl/cpu_cached_feature.py:31]: Looks good, better to provide example.

**Pull request merged**